sqlmesh/dbt/model.py

```diff
@@ -232,6 +232,9 @@ def model_kind(self, context: DbtContext) -> ModelKind:
                 valid_to_name="dbt_valid_to",
                 updated_at_name=self.updated_at,
                 updated_at_as_valid_from=True,
+                time_data_type=exp.DataType.build("TIMESTAMPTZ")
+                if target.type == "bigquery"
+                else exp.DataType.build("TIMESTAMP"),
```

Comment:
Just doing this for BQ for now until we can verify what the expected behavior is for other adapters.